File: nodejs-tools/nodejs-compiler-tools/formatter-xml/src/format_xml_element_attribute.ts
import { XmlElementAttribute } from '@vighnesh153/parser-xml';

export function formatXmlElementAttribute({ namespaces, value }: XmlElementAttribute): string {
  const combinedNamespace = namespaces.map((namespace) => namespace.tokenLiteral).join(':');
  return `${combinedNamespace}="${value.tokenLiteral}"`;
}

File: jvm-tools/XmlFormatter/src/main/kotlin/dev/vighnesh153/xml/formatter/fmt/FormatXmlElementAttribute.kt
package dev.vighnesh153.xml.formatter.fmt

import dev.vighnesh153.xml.formatter.ast.XmlElementAttribute

internal fun XmlElementAttribute.format(): String {
    val combinedNamespaces = namespaces.joinToString(":") { it.tokenLiteral }
    return """
        $combinedNamespaces="${value.tokenLiteral}"
        """.trim()
}